src/tokens/TokenImport.tsx

```diff
@@ -0,0 +1,195 @@
+import { useDispatch } from 'react-redux'
+import { showMessage } from 'src/alert/actions'
+import { AssetsEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import BackButton from 'src/components/BackButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import InLineNotification, { Severity } from 'src/components/InLineNotification'
+import TextInput from 'src/components/TextInput'
+import CustomHeader from 'src/components/header/CustomHeader'
+import GreenLoadingSpinner from 'src/icons/GreenLoadingSpinner'
+import { noHeader } from 'src/navigator/Headers'
+import { navigateBack } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { PasteButton } from 'src/tokens/PasteButton'
+import networkConfig from 'src/web3/networkConfig'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.TokenImport>
+
+export default function TokenImportScreen(_: Props) {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const [tokenAddress, setTokenAddress] = useState('')
+  const [tokenSymbol, setTokenSymbol] = useState('')
+  const [networkName, setNetworkName] = useState(NETWORK_NAMES[networkConfig.defaultNetworkId])
+
+  const navigateBackAndToast = () => {
+    ValoraAnalytics.track(AssetsEvents.import_token_submit, {
+      tokenAddress,
+      tokenSymbol,
+      networkId: networkConfig.networkId,
+    })
+    navigateBack()
+    // TODO RET-891: do this only when actually imported
+    dispatch(showMessage(t('tokenImport.importSuccess', { tokenSymbol })))
+  }
+
+  const errorMessage = () => {
+    // TODO RET-892: when states and validation are added, choose appropriate error or return null
+    const errors = [
+      t('tokenImport.error.invalidToken'),
+      t('tokenImport.error.alreadySupported'),
+      t('tokenImport.error.alreadyImported'),
+    ]
+    return <Text style={styles.errorLabel}>{errors[0]}</Text>
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <CustomHeader
+        style={styles.customHeader}
+        left={<BackButton />}
+        title={<Text style={styles.headerTitle}>{t('tokenImport.title')}</Text>}
+      />
+      <ScrollView>
+        <View style={styles.titleContainer}>
+          <InLineNotification
+            severity={Severity.Informational}
+            description={t('tokenImport.notification')}
+          />
+
+          <View style={styles.inputContainer}>
+            {/* Token Address */}
+            <View style={styles.textInputGroup}>
+              <Text style={styles.label}>{t('tokenImport.input.tokenAddress')}</Text>
+              <TextInput
```

Comment:
one thing i thought about while reading this, is the number of props that are repeated with each TextInput. it seems like we'd want to apply the same props consistently throughout this screen, and if we want to change / add / remove props, we'd need to do it in all 3 places. classic signals that we might want to componentise this - what do you think about an input component that includes the View, label, and text input? the props could be something like label, children, ...textInputProps.

Reply:
Agree, I think the code refactoring rules of thumb are good to follow:
- WET: Write Everything Twice
- Rule of three: once duplication repeats three times, do abstraction

Slight concerns that the network component is not editable and that it will likely be a dropdown list in the future.

I'll add a render function to abstract this.